raiden/api/rest.py

```diff
@@ -562,6 +562,15 @@ def open(
             settle_timeout=settle_timeout,
         )
         try:
+            token = self.raiden_api.raiden.chain.token(token_address)
+            balance = token.balance_of(self.raiden_api.raiden.address)
+            if total_deposit > balance:
```

Comment:
I suggest to change the code you added from what you have in the PR to:

```python
    if total_deposit is not None and total_deposit > balance:
        error_msg =  'Not enough balance to deposit. {} Available={} Needed={}'.format(
            pex(token_address),
            balance,
            total_deposit,
        )
        return api_error(
            errors=error_msg,
            status_code=HTTPStatus.PAYMENT_REQUIRED,
        )
```

1. The reason is that `total_deposit` can also be `None` in which case you don't need to check anything.
2. To return an error from the API you need to use `return api_error()`